src/App.tsx

```diff
@@ -1,41 +1,72 @@
+import { Post } from './types/Post';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [visiblePosts, setPosts] = useState<Post[]>([]);
+  const [selectedUserId, selectUser] = useState<number>(0);
+  const [selectedPostId, selectPost] = useState<number>(0);
+
+  const getPosts = async () => {
+    try {
+      const posts = await getUserPosts(selectedUserId);
+
+      setPosts(posts);
+    } catch {
+      throw new Error('Data was\'t found');
+    }
+  };
+
+  useEffect(() => {
+    getPosts();
+  }, [selectedUserId]);
+
+  return (
+    <div className="App">
+      <header className="App__header">
+        <label htmlFor="user">
+          Select a user: &nbsp;
+
+          <select
+            className="App__user-selector"
+            id="user"
+            value={selectedUserId}
+            onChange={(event) => selectUser(Number(event.target.value))}
+          >
+            <option value="0">All users</option>
+            <option value="1">Leanne Graham</option>
+            <option value="2">Ervin Howell</option>
+            <option value="3">Clementine Bauch</option>
+            <option value="4">Patricia Lebsack</option>
+            <option value="5">Chelsey Dietrich</option>
+            <option value="6">Mrs. Dennis Schulist</option>
+            <option value="7">Kurtis Weissnat</option>
+            <option value="8">Nicholas Runolfsdottir V</option>
+            <option value="9">Glenna Reichert</option>
+            <option value="10">Leanne Graham</option>
+          </select>
+        </label>
+      </header>
+
+      <main className="App__main">
+        <div className="App__sidebar">
+          <PostsList
+            posts={visiblePosts}
+            selectedPostId={selectedPostId}
+            selectPost={selectPost}
+          />
+        </div>
+
+        <div className="App__content">
+          {selectedPostId > 0 && <PostDetails selectedPostId={selectedPostId} />}
```

Comment:
0 will convert to false

```suggestion
          {selectedPostId && <PostDetails selectedPostId={selectedPostId} />}
```